### PR TITLE
TABMWP FOR REACT

### PR DESCRIPTION
**Pull request by GaoDalie**
### 🤔 Reasoning

_Explain the purpose of this PR..._

### 🚧 Changes

_Describe the changes made..._

### ✅ PR Checklist

- [x] Using this PR template?
- [ ] Linked issue?
- [ ] Added feature?
    - [ ] Added/updated docs?
    - [ ] Added/updated tests?

### CODE REVIEW

File: agential/cog/prompts/react.py [hunk 1]
```diff
@@ -232,3 +232,94 @@
...
 Thought 4: The evidence shows North Carolina entered Union in 1789, and Rhode Island entered Union in 1790, which is a year after North Carolina. 
 Action 4: Finish[Rhode Island]"""
+
+REACT_INSTRUCTION_TABMWP = """Read the following table and question to solve the problem,providing Observation that SUPPORTS the answer, or indicating if there is NOT ENOUGH INFORMATION. Action can be two types:
+(1) Python[Question], generates an executable python python code using the Sympy library using the Context. The first line of the code should be 'from sympy import *'
+(2) Finish[answer], which returns the answer and finishes the task.
+You have a maximum of {max_steps} steps.
+Here are some examples:
+{examples}
+(END OF EXAMPLES)
```
Comment (alckasoc) [inline agential/cog/prompts/react.py:242]: needs to be rewritten

File: agential/cog/prompts/react.py [hunk 2]
```diff
@@ -232,3 +232,94 @@
...
+Action 1 : Python['Calculate the total cost for Tracy by multiplying the price per kilogram of spherical beads, star-shaped beads, and flower-shaped beads by the respective quantities purchased and then summing these values.']
+Observation 1: [code python] # Recalculating the total cost for Tracy with provided prices and quantities to ensure accuracy price_spherical_beads = 3.42 price_star_shaped_beads = 1.95 price_flower_shaped_beads = 2.18 quantity_spherical = 5 quantity_star_shaped = 4 quantity_flower_shaped = 3 # Calculate total cost again total_cost = (price_spherical_beads * quantity_spherical) + \ (price_star_shaped_beads * quantity_star_shaped) + \ (price_flower_shaped_beads * quantity_flower_shaped) total_cost
+Thought 2:The calculation provides a straightforward method to find the total spending based on the quantities and prices of the beads.
+Action 2 : Finish[$31.44]
+"""
+
+Prompt_PG = """ You're a skilled Python programmer, Read the following mathematical question and then write Python code using the below python libraries to answer the question or generate an intermediate result. Information on each of the libraries is provided below.
```
Comment (alckasoc) [inline agential/cog/prompts/react.py:314]: follow naming conventions

File: agential/cog/functional/react.py
Comment (alckasoc) [inline agential/cog/functional/react.py:1]: don't change the default values in the parameters

File: agential/cog/agent/react.py
Comment (alckasoc) [inline agential/cog/agent/react.py:1]: dont change default values in parameters

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: agential/cog/prompts/react.py (hunk 1 above)
Comment (GaoDalie) [reply agential/cog/prompts/react.py]: I got all my prompt instructions from this repo where they build the React prompt for TABMWP. Here are all my references:

https://github.com/rakutentech/MathSensei/blob/main/src/REACT_prompts/prompts_MATH_react.py

https://github.com/rakutentech/MathSensei/blob/986c21528ad7b5719dc19f158cce0b9f2e122e60/src/custom_prompts/prompt_pg.py#L172"
-

**Pull request closed without merging**